gateway-js/src/index.ts

```diff
@@ -634,8 +634,8 @@ export class ApolloGateway implements GatewayInterface {
   }
 
   private createSchemaFromSupergraphSdl(supergraphSdl: string) {
-    const assumeValid = !(this.config.validateReceivedSupergraphs ?? process.env.NODE_ENV !== 'production');
-    const supergraph = Supergraph.build(supergraphSdl, { assumeValid });
+    const validateSupergraph = this.config.validateSupergraph ?? process.env.NODE_ENV !== 'production';
```

Comment:
I think this is wrong. If they explicitly set `validateSupergraph` to false, we should probably honor that, no?

Reply:
`false ?? true` equates to false, unlike `false || true`, so I think this is doing what we want.

Reply:
Yeah, of course. Clearly been doing too much rust lately. ship it.